src/cdk/v2/destinations/gladly/utils.js

```diff
@@ -0,0 +1,175 @@
+const get = require('get-value');
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const {
+  base64Convertor,
+  getDestinationExternalID,
+} = require('../../../../v0/util');
+const { MappedToDestinationKey } = require('../../../../constants');
+
+const reservedCustomAttributes = [
+  'email',
+  'phone',
+  'address',
+  'name',
+  'avatar',
+  'firstName',
+  'lastName',
+  'userId',
+];
+
+const externalIdKey = 'context.externalId.0.id';
+const identifierTypeKey = 'context.externalId.0.identifierType';
+
+const getHeaders = (destination) => {
+  const { apiToken, userName } = destination.Config;
+  const credentials = `${userName}:${apiToken}`;
+  const base64Credentials = base64Convertor(credentials);
+  return {
+    'Content-Type': 'application/json',
+    Authorization: `Basic ${base64Credentials}`,
+  };
+};
+
+const getEndpoint = (destination) => {
+  const { domain } = destination.Config;
+  return `https://${domain}/api/v1/customer-profiles`;
+};
+
+
+const getFieldValue = (field) => {
+  if (field) {
+    if (Array.isArray(field)) {
+      return field.map((item) => ({ original: item }));
+    }
+    return [{ original: field }];
+  }
+  return undefined;
```

Comment:
The customer can send a single email/phone as a string as well.

Reply:
line no 44 handles that scenario

Reply:
Yes my bad